src/typescript/frontend/src/hooks/use-window-dimensions.ts

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from "react";
+
+function getWindowDimensions() {
+  const { innerWidth: width, innerHeight: height } = window;
+  return {
+    width,
+    height,
+  };
+}
+
+export default function useWindowDimensions() {
+  const [windowDimensions, setWindowDimensions] = useState(getWindowDimensions());
+
+  useEffect(() => {
+    function handleResize() {
+      setWindowDimensions(getWindowDimensions());
+    }
+
+    window.addEventListener("resize", handleResize);
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+
+  return windowDimensions;
+}
```

Comment:
```suggestion
```

Appreciate the custom hook, but we have `react-use` already so you can just use this 😄 

```react
import { useWindowSize } from "react-use";
const { width, height } = useWindowSize();
```